routing/payment_lifecycle.go

```diff
@@ -673,6 +678,18 @@ func (p *paymentLifecycle) sendAttempt(
 		PaymentHash: *attempt.Hash,
 	}
 
+	buffer := new(bytes.Buffer)
+	if err := p.firstHopTLVs.Encode(buffer); err != nil {
+		return p.failAttempt(attempt.AttemptID, err)
+	}
+
+	recordsBytes := buffer.Bytes()
+	tlvRecord := tlv.NewPrimitiveRecord[lnwire.CustomRecordsBlobTlvType](
+		recordsBytes,
+	)
+
+	htlcAdd.CustomRecordsBlob = tlv.SomeRecordT(tlvRecord)
```

Comment:
So I chatted with @ffranr about this offline, but instead we only want to be setting `ExtraData`. All the other logic (further down in the state machine), will _only_ look inside of `ExtraData`, as it isn't expecting that extra layer of wrapping. 

We _do_ want the upfront validation here tho to ensure that w/e we're passed in is actually a canonical TLV stream.